6_visualize/src/prep_spark_line_fun.R

```diff
@@ -0,0 +1,71 @@
+
+prep_spark_line_fun <- function(storm_data_ind_file, viz_config_yaml, DateTime){
+  storm_data <- readRDS(sc_retrieve(storm_data_ind_file))
+  viz_config <- yaml::yaml.load_file(viz_config_yaml)
+
+  sites <- unique(storm_data$site_no)
+  this_DateTime <- as.POSIXct(DateTime, tz = "UTC") # WARNING, IF WE EVER MOVE FROM UTC elsewhere, this will be fragile/bad.
+  this_spark <- filter(storm_data, dateTime <= this_DateTime) # keep all data up until this timestep
+
+  plot_fun <- function(){
+    orig_par <- par()
+
+    # Set spacing configurations
+    x_coords <- c(viz_config$sparks$xleft, viz_config$sparks$xright)
+    y_start <- viz_config$sparks$ybottom
+    y_space <- viz_config$sparks$ytop-y_start
+    vertical_spacing <- y_space/length(sites)
+
+    y_pos <- y_start # initialize position of first spark
+    for(i in sites) {
+
+      # Filter data to just one site & create polygon out of it
+      storm_data_i <- filter(this_spark, site_no == i) %>% arrange(dateTime)
+      storm_data_site_all <- filter(storm_data, site_no == i)
+
+      if(nrow(storm_data_i) == 0) { next } # skip if there's no data at or before this timestep
+
+      flood_stage_va <- as.numeric(unique(storm_data_i$flood_stage_normalized))
+      lowest_stage <- min(storm_data_site_all$stage_normalized, na.rm = TRUE)
+      highest_stage <- max(storm_data_site_all$stage_normalized, na.rm = TRUE)
+
+      full_polygon <- data.frame(dateTime = head(storm_data_i$dateTime, 1),
+                                 stage_normalized = min(storm_data_i$stage_normalized,1)) %>%
+        bind_rows(select(storm_data_i, dateTime, stage_normalized)) %>%
+        bind_rows(data.frame(dateTime = tail(storm_data_i$dateTime, 1),
+                             stage_normalized = min(storm_data_i$stage_normalized)))
+
+      # Fill values lower than flood stage with the stage & then create a polygon out of it
+      above_flood_data <- mutate(storm_data_i, stage_normalized = ifelse(stage_normalized < flood_stage_va,
+                                                                         flood_stage_va,
+                                                                         stage_normalized))
```

Comment:
this `ifelse` statement is equivalent to `pmax(stage_normalized, flood_stage_va)`, in case you like that better. (i do but don't feel strongly about it)

Reply:
Ooooo I just didn't realize that was a function. Wow - game changer! I do this ifelse construction a lot....thanks!